src/models/geograph.py

```diff
@@ -52,7 +54,7 @@ class GeoGraph:
     def __init__(
         self,
         data,
-        crs: Union[str, pyproj.CRS],
+        crs: Union[str, pyproj.CRS] = None,
```

Comment:
Interesting question,
the probably most extensive answer is the [class definition here](https://pyproj4.github.io/pyproj/dev/api/crs/crs.html), but that's likely not what you mean by the question I guess.

The main reason for allowing `str` and `pyproj.CRS` here is that this is the underlying way in which `geopandas` stores and can accept CRS information (c.f. [here](https://geopandas.org/docs/user_guide/projections.html) - if you give a string that's essentially parsed into a `pyproj.CRS` by geopandas under the hood). 
Strings for CRS are useful because they allow you to use generally defined CRS (like the UTM ones, or WGS84, ...). But if you were to do your custom projection of the sphere you would get your own CRS which does not have a canonical string name. For such use cases, where users work with their custom projections, I wanted to also allow `pyproj.CRS`. In terms of handling everything it doesn't change much, as geopandas does all of the CRS stuff for us 😄 

[Pyproj](https://pyproj4.github.io/pyproj/dev/index.html) is  the canonical CRS handling library in python that uses the C++ library [PROJ](https://proj.org/about.html) under the hood. `Rasterio` uses it's [own **binder](https://github.com/mapbox/rasterio/blob/master/rasterio/crs.py) to the PROJ library, rather than pyproj. Since they both go to the underlying PROJ tough, they interface nicely as far as I know. 

As for `ipyleaflets` I'm less familiar, they seem to be using [dictionaries](https://github.com/jupyter-widgets/ipyleaflet/blob/38a7e9d8284aa1c3d7fc465415569e312a387be6/ipyleaflet/leaflet.py#L1970) (see also [this sample notebook with custom projection](https://github.com/jupyter-widgets/ipyleaflet/blob/38a7e9d8284aa1c3d7fc465415569e312a387be6/examples/CustomProjections.ipynb)). They don't include `pyproj` in their dependencies, so they likely have their own binder to PROJ as well (as rasterio does). I would be surprised if they didn't use PROJ though, since this seems to be the defacto standard since ~30 yrs in the GIS community. 

So bottom line to your question: I think deep under the hood they are all the same (PROJ), but the interfaces may vary.